### PR TITLE
TCM-381 | Button State Login

### PR DESCRIPTION
**Pull request by vespertineSiren**
### Summary

- Enable/disable button based on contents in InputView
- Create function that enables button state and switches color based on two static values. 

### Issue

[TCM-381](https://hyfnla.atlassian.net/browse/TCM-381)

### Video/Screenshot

| Initial | Filled |
|----|----|
|initial|filled|

https://user-images.githubusercontent.com/27742105/119737184-b7ba9500-be33-11eb-9dac-f5a2c0b5b590.mov

### CODE REVIEW

File: app/src/main/java/com/trx/consumer/common/CommonButton.kt [hunk 1]
```diff
@@ -44,4 +45,11 @@ class CommonButton @JvmOverloads constructor(
     fun textColor(color: Int) {
         this.setTextColor(ContextCompat.getColor(this.context, color))
     }
+
+    fun setPrimaryEnabled(isEnabled: Boolean) {
+        if (isEnabled != isEnabled()) {
+            bgColor(if (isEnabled) R.color.black else R.color.greyDark)
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/common/CommonButton.kt:51]: Not sure we want specific colors built into a reusable view, would be better as an extension function

Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/common/CommonButton.kt:51]: I thought it through that way but I was recommended to code it this way initially. Though that is a good suggestion for a refactor if given the greenlight in the future.

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Just one thought

**Review by VivekV95 — APPROVED** [review]

File: app/src/main/java/com/trx/consumer/common/CommonButton.kt (hunk 1 above)
Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/common/CommonButton.kt:51]: Confirmed with @onodude, let's make this an extension

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Just the extension

File: app/src/main/java/com/trx/consumer/common/CommonButton.kt (hunk 1 above)
Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/common/CommonButton.kt:51]: Cool will inline.

**Review by onodude — APPROVED** [review]
thank you!

**Pull request merged**